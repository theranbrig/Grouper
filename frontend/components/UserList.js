import gql from 'graphql-tag';
import React, { Component } from 'react';
import { Mutation } from 'react-apollo';
import { Button, List, Segment } from 'semantic-ui-react';
import { INDIVIDUAL_LIST_QUERY } from './List';

const REMOVE_USER_MUTATION = gql`
	mutation REMOVE_USER_MUTATION($id: ID!, $email: String!) {
		removeUser(id: $id, email: $email) {
			id
		}
	}
`;

export default class UserList extends Component {
	render() {
		return (
			<List inverted relaxed>
				{this.props.list.users.map(user => (
					<List.Item key={user.id}>
						<Segment inverted textAlign="left">
							<Mutation
								mutation={REMOVE_USER_MUTATION}
								variables={{ id: this.props.id, email: user.email }}
								refetchQueries={[
									{ query: INDIVIDUAL_LIST_QUERY, variables: { id: this.props.id } }
								]}>
								{(removeUser, { error, loading }) => {
									if (error) <p>Error...</p>;
									return (
										<List.Content floated="right" verticalAlign="middle">
											<Button
												basic
												color="orange"
												icon="close"
												onClick={async e => {
													e.preventDefault();
													removeUser();
												}}
											/>
										</List.Content>
									);
								}}
							</Mutation>
							<List.Header>{user.username}</List.Header>
							<List.Description>{user.email}</List.Description>
						</Segment>
					</List.Item>
				))}
			</List>
		);
	}
}
