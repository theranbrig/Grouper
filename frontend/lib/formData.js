const typeOptions = [
  { key: 'g', text: 'General', value: 'General' },
  { key: 'gi', text: 'Gift', value: 'Gift' },
  { key: 'gr', text: 'Groceries', value: 'Groceries' },
  { key: 'm', text: 'Market', value: 'Market' },
  { key: 'p', text: 'Party', value: 'Party' },
  { key: 's', text: 'Shopping', value: 'Shopping' },
];

export default typeOptions;
