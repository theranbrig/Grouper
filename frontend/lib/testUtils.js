import casual from 'casual';

casual.seed(777);

const fakeUser = () => ({
  __typename: 'User',
  id: '4234',
  username: casual.name,
  email: casual.email,
  password: 'password',
  permissions: ['USER'],
});

const fakeItem = () => ({
  __typename: 'ListItem',
  id: '3534',
  name: 'Habaneros',
  price: 12,
  list: '1234567890',
  user: fakeUser(),
  inCart: false,
});

const fakeList = () => ({
  __typename: 'List',
  id: '01234567890',
  users: [fakeUser()],
  name: 'Grocery List',
  type: 'Groceries',
  items: [fakeItem()],
});

export { fakeList, fakeUser, fakeItem };
