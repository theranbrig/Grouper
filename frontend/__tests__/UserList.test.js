import { mount, shallow } from 'enzyme';
import wait from 'waait';
import toJSON from 'enzyme-to-json';
import UserList from '../components/UserList';
import { MockedProvider } from 'react-apollo/test-utils';
import { fakeUser, fakeItem } from '../lib/testUtils';

const listData = {
	id: '01234567890',
	users: [fakeUser()],
	name: 'Grocery List',
	type: 'Groceries',
	items: [fakeItem()]
};

describe('<Login/>', () => {
	it('should render the login page', () => {
		const wrapper = shallow(<UserList list={listData} />);
		expect(toJSON(wrapper.find('h2'))).toMatchSnapshot();
	});
});
