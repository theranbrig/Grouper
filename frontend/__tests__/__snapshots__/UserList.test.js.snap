// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Login/> should render the login page 1`] = `null`;
